src/subscription_manager/gui/registergui.py

```diff
@@ -124,7 +124,130 @@ def reset_resolver():
+            has been called and a callback registered. If that's the case,
+            then RegisterScreen._set_screen() sets the current screen to a progress screen.
+
+        The return value of RegisterScreen.run_pre() is ignored, and
+        RegisterScreen.register() returns False.
+
+        This returns to rhsm_log.apply(), where valid_registration is
+        set to the return value. valid_registration=True indicates a
+        succesful registration
+
+
+        if valid_registration=True, we are basically done with register.
+        But rhsm_login can't return from apply() yet, since that could
+        potential lead to firstboot ending if it's the last or only module.
+
+        gtk main loop iterations are run, mostly to let any threads finish
+        up and any idle loop thread watchers to dry up.
+
+
+        The return value of rhsm_login.apply() at this point is actualy
+        the _apply_result instance variable. register Screens are expected
+        to set this by calling their _a_ finish_registration() method. For
+        subscription-manager-gui that means RegisterScreen.finish_registration,
+        usually access as a Screens self._parent.finish_registration.
+
+        For firstboot screens, self._parent will be rhsm_module.moduleClass
+        (also a subclass of RegisterScreen).
+
+        rhsm_module.finish_registration() will check the "failed" boolean,
+        and either return to a Screen (CredentialsPage, atm). Or if
+        failed=True, it will also call RegisterScreen.finish_registration(),
+        that closes the gui window.
+
+        The UI flow is a result of the order of RegisterScreen._screens,
+        and the screen indexes returned by Screen.apply().
+
+        But, between the Screen activity call also change the flow, most
+        notably the results of any async calls and callbacks invoked from
+        the screens .pre()
+
+        A common case is the async callbacks error handling calling
+        self._parent.finish_registration(failed=True)
+
+        async callback can also call RegisterScreen.pre_done() to send the
+        UI to a different screen. rhsm api calls results that indicate multiple
+        choices for a sub would send flow to a chooseSub GuiScreen vs a
+        NonGuiScreen for attaching a sub, for example.
+
+        RegisterScreen.run_pre schedules async jobs, they get queue, and
+        wait for there callbacks. The callbacks then can use pre_done()
+        to finish the tasks the run_pre started. Typicaly the UI will
+        see the Progress screens in the meantime.
+
+        If going to screen requires a ayncs task, run_pre starts it by
+        calling the new screens pre(), setting that screen to current (_set_screen),
+        and then setting the GuiScreen to the progress screens. Screen
+        transitions that don't need async tasks just return nothing from
+        there pre() and goes to  next screen in order in self._screegoes to
```

Comment:
their pre() and go to the next screen in the order.

Not sure what the rest is supposed to mean.